Source/SDK/ClientBuilder.cs

```diff
@@ -158,6 +165,25 @@ public ClientBuilder WithCancellation(CancellationToken cancellation)
             return this;
         }
 
+        /// <summary>
+        /// Registers all artifacts by scanning all assemblies.
+        /// </summary>
+        /// <param name="assemblyFilter">The filter for assemblies.</param>
+        /// <param name="includeExeFiles">Whether or not to also load assemblies from .exe files.</param>
+        /// <returns>The client builder for continuation.</returns>
+        public ClientBuilder WithAllArtifacts(Func<Assembly, bool> assemblyFilter = default, bool includeExeFiles = false)
+            => WithAllScannedAssemblies(
+                asm =>
+                {
+                    _embeddingsBuilder.RegisterAllFrom(asm);
+                    _projectionsBuilder.RegisterAllFrom(asm);
+                    _aggregateRootsBuilder.RegisterAllFrom(asm);
+                    _eventHandlersBuilder.RegisterAllFrom(asm);
+                    _eventTypesBuilder.RegisterAllFrom(asm);
+                },
+                assemblyFilter,
+                includeExeFiles);
```

Comment:
Oh, this is the fancy thing. Let's have a chat about the API here. It looks a little bit scary.